packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationModule.m

```diff
@@ -0,0 +1,222 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+#import <EXScreenOrientation/EXScreenOrientationModule.h>
+#import <UMCore/UMEventEmitterService.h>
+#import <UMCore/UMDefines.h>
+#import <UMCore/UMModuleRegistryProvider.h>
+#import <EXScreenOrientation/EXScreenOrientationUtilities.h>
+
+#import <UIKit/UIKit.h>
+
+@interface EXScreenOrientationModule ()
+
+@property (nonatomic, weak) id<UMEventEmitterService> eventEmitter;
+@property (nonatomic, assign) bool hasListeners;
+@property (nonatomic, assign) UIInterfaceOrientation currentScreenOrientation;
+@property (nonatomic, assign) UIInterfaceOrientationMask currentScreenOrientationMask;
+
+@end
+
+@implementation EXScreenOrientationModule
+
+UM_EXPORT_MODULE(ExpoScreenOrientation);
+
+- (instancetype)init {
+  if (self = [super init]) {
+    _currentScreenOrientationMask = 0;
+  }
+  
+  return self;
+}
+
+- (void)dealloc
+{
+  [[NSNotificationCenter defaultCenter] removeObserver:self];
+  
+  dispatch_async(dispatch_get_main_queue(), ^{
+    [[UIDevice currentDevice] endGeneratingDeviceOrientationNotifications];
```

Comment:
No, it shouldn't be there. We need this when the module is active to track current screen orientation.

Reply:
If so… maybe it should be the singleton who listens to device orientation changes? This could make sense given that there is only one singleton, as there is only one device, we want to react to device orientation changes even if native module is missing (for unknown reason). The singleton would then dispatch events to listeners. What do you think?